### ISSUE TITLE
.get on dict with optional value infers 'str | _T@get | None' and reports type error

### ISSUE THREAD

**Issue by tekumara** [microsoft/pyright#1947]
```python
from typing import Dict, List, Optional


def mydict() -> Dict[str, Optional[str]]:
    return {"a": "1", "b": None}


def get() -> Optional[str]:
    return mydict().get("c", None)
```

Errors with:

```
test.py:9:12 - error: Expression of type "str | _T@get | None" cannot be assigned to return type "str | None"
    Type "str | _T@get | None" cannot be assigned to type "str | None"
      Type "object" cannot be assigned to type "str | None"
        "object" is incompatible with "str"
        Type cannot be assigned to type "None" (reportGeneralTypeIssues)
```

I was expecting no type error given the following passes without a type error:
```
def get() -> Optional[str]:
    return mydict()["c"]
```

pyright 1.1.146

**Comment by erictraut** [comment]
The problem in this case was that the function signature used a union for input parameters and the return type, and a TypeVar was included in both. You passed `None`, which was already specified in the union, so the TypeVar didn't get assigned a value during the constraint solving phase, and it was left "unsolved".

Here's a simplified version:

```python
def test(x: str | None | _T) -> str | None | _T:
    ...

reveal_type(test(None)) # str | None | _T ?
```

I've modified the logic to eliminate unsolved TypeVars from unions at the end of the constrain solving process. This fixes the issue.

**Comment by erictraut** [comment]
This is included in pyright 1.1.147, which I just published. It will also be included in the next release of pylance.
